src/main/java/at/ac/tuwien/damap/rest/dmp/service/DmpService.java

```diff
@@ -234,6 +245,21 @@ private void updateProjectLead(Dmp dmp) {
         }
     }
 
+    private void fetchORCIDContributorInfo(Dmp dmp) {
+        dmp.getContributorList().forEach(contributor -> {
+            // Existing contributor? Do not fetch data.
+            if (contributor.id != null) {
+                return;
+            }
+
+            var identifier = contributor.getPersonIdentifier();
+            if (identifier != null
+                    && identifier.getIdentifierType().equals(EIdentifierType.ORCID)) {
+                ContributorDOMapper.mapDOtoEntity(orcidPersonService.read(identifier.getIdentifier()), contributor);
```

Comment:
If you create a contributor manually, like so:
<img width="979" alt="image" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/db546715-0f0b-4d6a-ad48-7475b3eeb1f7">
And try to save the dmp, it fails.
The problem is, that the PersonIdentifier != null, and the IdentifierType is ORCID. This leads to a get request in the OrcidPersonService with the orcid query parameter being "". This returns an uncaught 404 error.

Reply:
Thank you. Wrapped it in a general try/catch block.